Add SMTP driver

patrickdk77 opened the pull request:
Attempting to test this, but when I run make it removes changes to the crd file, not sure what I'm missing here.

taruntej-a committed:
feat: initial working version of endpoint-monitoring-operator with CRD, drivers (http, dns, trino, etc.), and Slack notifier (#1)

Added the initial version of working code.

iam404 committed:
Update README.md

iam404 committed:
Update README.md

iam404 committed:
Update typo

garlic-naan committed:
feat(notifier): add Discord notifier support

- Introduced DiscordConfig struct and added it to NotifyConfig in the API.
- Implemented DiscordNotifier in internal/notifier/discord.
- Registered Discord notifier in the NotifierFactory for dynamic instantiation.

garlic-naan committed:
chore(api): regenerate deepcopy methods and CRDs for DiscordConfig

garlic-naan committed:
Refactor notifier alert logic, enhance Discord messages, and improve test coverage

- Extract common alert decision logic () to internal/notifier/notifier.go for reuse across notifiers.
- Refactor both Discord and Slack notifiers to use the shared ShouldAlert function, reducing code duplication.
- Enhance Discord alert messages with advanced Markdown, emoji, and improved formatting for better readability in Discord channels.
- Update and expand tests in discord_test.go to verify the new message formatting and shared alert logic.

taruntej-a committed:
Merge pull request #6 from garlic-naan/feat/notifier-discord

Feat/notifier-discord

patrickdk77 committed:
Add SMTP driver

taruntej-a commented on examples/smtp.yaml:
```diff
@@ -0,0 +1,15 @@
+  name: smtp-check
+  namespace: endpoint-monitoring-operator-system
+spec:
+  driver: tcp
```
There's a mismatch between the example and the implementation. 
`driver: tcp`
But this should be:
`driver: smtp`
Since you've implemented the SMTP driver and registered it in factory.go (line 97), the example should reference smtp to actually use the new driver. Currently, this example would just perform a basic TCP connectivity check instead of the SMTP-specific validation that your driver provides.

patrickdk77 commented:
Yes but that would not explain why the crd file gets wiped out when I attempt to build this, so as I said, I have not been able to test it, due to the crd file having it's changes removed, and I do not understand why they are removed.

tarunteja483 commented:
I think I found the issue @patrickdk77  :

  The Problem

  - CRD YAML has typo: smartTls should be startTls?
  - Go types are correct: StartTls
  - Running make regenerates CRD from Go types (fixing the typo)

Commit:
chore: update Docker image to v1.0.1 with Discord notifier support

- Update image from tarun4licious to licioustech registry
- Bump version from 1.0.0 to 1.0.1
- Includes Discord notifier feature from PR #6

taruntej-a committed:
chore: update Docker image to v1.0.1 with Discord notifier support

chore: update Docker image to v1.0.1 with Discord notifier support

patrickdk77 committed:
Fix typo

patrickdk77 committed:
Add notifications for email and webhooks
- use secrets
- support templates
- add aws ses dependencies

patrickdk77 committed:
Merge branch 'upstream' into feat/email-notifier

patrickdk77 committed:
smtp driver changes

patrickdk77 committed:
Merge branch 'upstream' into feat/smtp-driver

patrickdk77 committed:
Merge branch 'feat/email-notifier' into feat/smtp-driver
Bring in shared dependencies from email-notifier
